### ISSUE TITLE
Redesigning site's layout

### ISSUE THREAD

**Issue by Tomasz-Kluczkowski** [python-sprints/python-sprints.github.io#83]
It's time to do some high level thinking of what goes where and what gets deleted.
Please dissect this and comment on each part. No rush as always, no pressure - but you asked for it :).

Based on your comment below:
"How I see it (and it's just an idea), it's that for it, the website should be a bit more functional. Meaning that instead of using the whole home page for messages selling the group (missing, what we do...), we could have a longer list of events, links to the documentation, may be a map with the meetups that host sprints (similar to the one in the pandas sprint)..."

# Home Page:

## Removing part
- Get rid of "our mission" section
- Get rid of "what we do section" (services) - move it to about page (if it will fit into the design).
- Get rid of current way upcoming event is displayed
- Get rid of recent events completely

## Adding part:
- At the top: **upcoming events** for each chapter shown as events are shown now in Events page (basically the way the header for the post looks like). We may need to redesign the layout a bit to see if we can fit the sponsor's logo of the event somewhere.
- Some kind of **"Chapters" section** where logos of cities are shown with the city name and user can click on that to see upcoming (or all) events for their city. This can also replace the above section so that the **user has to select the chapter first** before they see events (it all depends on the space we want the events to occupy, initially when we only have ours once / twice a month we have plenty of space and can show its details).
Example of a city logo:
<img src="http://www.bedehouse.org.uk/wp-content/uploads/2012/10/City-Bridge-Trust-City-of-London-logo.graphics.gif" width="200">
Here we could also have the google style map with geo-locators same as for pandas event.

- **Get Started section** - set of toggled titles with short descriptions of what kind of problem one can learn about in this toggle and links to the articles explaining them - we should already have some pull request  and pandas guides.

- If we get more sponsors and they will stop fitting in the footer / taking close to 90% of it - we need to move them all into **sponsors section** which should be at the bottom of the home page.
 
# About Page:

I will need to replace the shitty quality photos there and now since we want to make this sort of global I will use good photos from other chapters showing the diversity of our group. Will need to maybe ask people for a list of cities that want to associate with us.

# Get Started page:
- To be removed completely

# Events page:
- Renamed to Past Events? Should show chapters to select from and then past events per chapter once user selects one.

# Contacts Page:
- Stays as it is.

# Credits Page:
- To be added as the last part of work. Place to say thanks to people who's free of charge work we are using and abusing as well as mention our own names (bragging rights, right ? :)).

Ok, thats enough material to tire your brain on. Have a look, let me know what you think when we see each other on Tuesday.
I will get this printed out so we can make notes on the side. 
Any other ideas - let me know.

**Comment by datapythonista** [comment]
![img_20180316_150954](https://user-images.githubusercontent.com/10058240/37528424-83776e7a-292c-11e8-87d5-b45e9eae02db.jpg)

For the home, something like this would make sense to me. I hope you can get the idea, with my Picasso-like drawing skills. ;)

Then, to me just a chapter page, and an even page would be needed. The chapter one could have:
- A picture
- Some short description
- The list of events (both future and past?)
- The sponsors
- The link to meetup

I quite like the layout of the "About us" page you have, I'd reuse at least the top part of that for the chapter page:

![lps](https://user-images.githubusercontent.com/10058240/37528961-d24af6d8-292d-11e8-8c6b-01aa6b12e92f.png)

For the events we discussed that.

Then, may be we could have a page with resources. Screencast on how to send a pull request, how to set up a conda environment... But that would be a simple static page, and we can add at a later time.

What do you think? Does it make sense to you? Anything I'm missing?

**Comment by Tomasz-Kluczkowski** [comment]
Hola! Señor Picasso! Thx for such a quick reply (honestly I thought that since its Friday the beer will win but you have proven me wrong)

I have some time tomorrow as I have to get my daughter to Saturday school and wait for her - I will start drafing this.

Lovely people figures in the about section :) you got some skill.
Ok more seriously - how I understand the new layout you created:

Home page:

I am not sure if I can fit the list of events to the left next to the map and a list of chapters to the right (not on mobiles) but it should work on desktops. For mobiles the „next events” section will be at the top, then „chapters” as I think the priority is to show what is happening first?

About Us - ok just to copy what there already is in the about page (without the top photo (use that for our chapter) and replace the low quality photos)? As I understand this would be about us for the entire organisation, not for just London Python Sprints.
Maybe make it a little bit more condensed?

Sponsors - as a separate section, removed from the footer.

Now the navigation problem. What subpages do we want:
Home
Chapters (or the selection is only available from the home page)
Resources
Credits? (Or just keep a href to credits in footer only...thats how I solved it on my personal blog)

In case we reduce the navigation too much (to say only one link) we would have to convert it to single page navigation, but then resources and everything else would have to be tacked in as well...

I can work both out but naturally everything costs time :).

Anyway, Im having a nice big Peroni Nastro Azzuro. Take care.

On 16 Mar 2018, at 15:26, Marc Garcia <notifications@github.com<mailto:notifications@github.com>> wrote:
[37528424-83776e7a-292c-11e8-87d5-b45e9eae02db.jpg.jpeg]


For the home, something like this would make sense to me. I hope you can get the idea, with my Picasso-like drawing skills. ;)

Then, to me just a chapter page, and an even page would be needed. The chapter one could have:

  *   A picture
  *   Some short description
  *   The list of events (both future and past?)
  *   The sponsors
  *   The link to meetup

I quite like the layout of the "About us" page you have, I'd reuse at least the top part of that for the chapter page:

[lps]<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F10058240%2F37528961-d24af6d8-292d-11e8-8c6b-01aa6b12e92f.png&data=02%7C01%7C%7Cca6a050a9d4b443f70da08d58b524e07%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636568107961979174&sdata=ZPwA5e8pNSo3%2FRFCjQ%2BD8PFFeNy0KrtmEOBae1A84Fs%3D&reserved=0>

For the events we discussed that.

Then, may be we could have a page with resources. Screencast on how to send a pull request, how to set up a conda environment... But that would be a simple static page, and we can add at a later time.

What do you think? Does it make sense to you? Anything I'm missing?

**Comment by datapythonista** [comment]
All of your proposals sounds good. In the pandas sprint page the two columns render as one in the mobile version. It's using bootstrap to make it responsive, not sure if you're using it.

I would forget the resources page for now, we don't have any resource I think it's worth sharing at the moment, so I wouldn't overthink it.

The credits is mainly for you. I'm happy to have it in the footer or as a separate page. I'd have it in the footer, as I wouldn't expect anyone to click on the link, and you deserve that people know you did all the work. :) I'm happy to keep the sponsors in the footer, I draw them in a separate section for no particular reason, probably it's not worth.

Don't take my beautiful drawing as a closed spec, it was just to show what I had in my mind. I think it makes sense that people can quickly see what the page is about, and quickly go to the events or the specific chapter, so those are the two key parts (about us, and list of chapters/events), but feel free to do it the way it makes sense to you.

**Comment by Tomasz-Kluczkowski** [comment]
- I am using bootstrap 4 so it should be the case of transferring and adapting.

- I hope nobody will mind if I reuse the code you already have for map / locations. I will see if I can make adding new geo-locators to the map a bit more user-friendly (and keep the chapters location as yaml data contained in the actual chapter_name.md file, then pull it and add as a javascript variable - if I can).

 
- resources -  that's the last bit. Home page first and foremost.

- ok, sponsors in the footer, link to credits in the footer **only** as well

**Comment by datapythonista** [comment]
Not sure how many credits you have, but if it's you and my brother, what I meant is that you could have the credits directly in the footer, not a link to a separate page. I have no problem with a separate page, but I don't think anyone will click on the credits link, so you may want to be directly in the footer. Just an idea, it's your call.

**Comment by Tomasz-Kluczkowski** [comment]
I am building automatic location markers for the chapters map slowly.
The issue I see is this:
What should this map really show? (There is TLDR at the bottom :)).

I could show each chapter and their “permanent” (if there is such a thing) address - as an example for us it would be:
London Python Sprints
@Harvey Nash
110 Bishopsgate
Etc etc.
Uk

Since we have more than one sponsor and other chapters can also have multiple - I cannot thing of sticking all sponsors on the map locators.
We have to choose one address where we meet most I think - and same for all other chapters.

Alternatively we can just have a general location of the city where each chapter is without any address (just a latitude/ longitude of the city centre).

Or we only want to show locations of next events - then it would be much less locators but we could have it exactly as it was for pandas. But then if a particular chapter has no event - it does not get shown on the map at all...

TLDR:

1. Show all chapters that signed up with one “permanent” address of their selection.

2. Show all chapters without address - only lat/lng of their city centre.

3. Only show next events locations with the appropriate address but chapters currently not running any events do not get shown on the map at all.

I think that below the map I would make a text list of all chapters (clickable buttons or just simple href’s) anyway as you suggested on the drawing. It makes sense. Maybe even little city icons could be used to make them more appealing graphically. So even if they are not displayed on the map if they run no events - people could click on them and get into chapter’s detail view page.
On 17 Mar 2018, at 19:15, Marc Garcia <notifications@github.com<mailto:notifications@github.com>> wrote:


Not sure how many credits you have, but if it's you and my brother, what I meant is that you could have the credits directly in the footer, not a link to a separate page. I have no problem with a separate page, but I don't think anyone will click on the credits link, so you may want to be directly in the footer. Just an idea, it's your call.

**Comment by datapythonista** [comment]
I'd simply have one marker per chapter in the city center.

**Comment by datapythonista** [comment]
Btw, do you want to talk 5 minutes about the website at the next meetup? That would be great, may be more people starts contributing to it. :)

**Comment by Tomasz-Kluczkowski** [comment]
Sure, np. I have a nice list of jobs which I could seriously use others’ help with. I shall try to gather my thoughts up and print something out and will see if anyone has any clue how to solve those issues.

Now that it no longer is being built for London only maybe people will see the bigger picture :).

Good idea.


On 18 Mar 2018, at 21:45, Marc Garcia <notifications@github.com<mailto:notifications@github.com>> wrote:


Btw, do you want to talk 5 minutes about the website at the next meetup? That would be great, may be more people starts contributing to it. :)

**Comment by datapythonista** [comment]
Cool. If you have some small tickets created for the sprint, I'm sure we'll find someone who prefers them over pandas. :)
